### ISSUE TITLE
MAUI: Can't rebuild the app after installing Plugin.Firebase package

### ISSUE THREAD

**Issue by sreejith-pagematics** [TobiasBuchholz/Plugin.Firebase#231]
I can't rebuild the app in release mode for creating APK files after the installation of `Plugin.Firebase` and `Plugin.Firebase.Crashlytics`. If I remove these packages I can rebuild and create APK file.

I follow [this](https://cedricgabrang.medium.com/firebase-push-notifications-in-net-maui-android-32c808844d7e) blog and they already suggesting to install 2 packages to prevent build errors. I did that too, still getting a lot of errors.

![a](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/33147ca9-b993-41e7-b86b-a80b54063d39)

**My Error screenshot:**

![image_2023_11_15T14_26_59_884Z](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/e2075bb2-fe1a-4425-a021-5a3cbea35c13)

![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/8c025dc8-7a2f-4901-93f6-643fbad4ae73)

> Severity Code Description Project File Line Suppression State Error NU1605 Warning As Error: Detected package downgrade: Xamarin.Kotlin.StdLib.Jdk8 from 1.7.20.1 to 1.7.10. Reference the package directly from the project to select a different version. MyApp -> Xamarin.AndroidX.Navigation.Runtime 2.5.2.1 -> Xamarin.AndroidX.Lifecycle.Runtime.Ktx 2.5.1.1 -> Xamarin.KotlinX.Coroutines.Android 1.6.4.1 -> Xamarin.Kotlin.StdLib.Jdk8 (>= 1.7.20.1) MyApp -> Xamarin.Kotlin.StdLib.Jdk8 (>= 1.7.10) MyApp E:\My Projects\MAUI\listpm-requester-app-maui\MyApp\MyApp.csproj 1

> Severity Code Description Project File Line Suppression State Error MSB3030 Could not copy the file "C:\Users.nuget\packages\xamarin.firebase.ios.cloudfirestore\8.10.0.3\lib\net6.0-ios15.4\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64_x86_64-simulator\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\core\http_uri.upb.h" because it was not found. MyApp C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets 4879

Is it a known issue? Any solutions?

**Comment by TobiasBuchholz** [comment]
I've recently released a new version of the plugin which slightly changed the setup, see the [readme](https://github.com/TobiasBuchholz/Plugin.Firebase#android-specifics). So instead of adding the mentioned packages from above you'll need to add the following:

```xml
<ItemGroup Condition="'$(TargetFramework)' == 'net7.0-android'">
    <PackageReference Include="Xamarin.AndroidX.Core" Version="1.12.0.2" />
    <PackageReference Include="Xamarin.AndroidX.Collection" Version="1.3.0.1" />
    <PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.3.0.1" />
    <PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.0.1" />
</ItemGroup>
```

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I have added the mentioned packages and tried to install the latest version, but getting below error.

> Severity	Code	Description	Project	File	Line	Suppression State
Error		Package restore failed. Rolling back package changes for 'MyApp'.				

> Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1605	Warning As Error: Detected package downgrade: Xamarin.Google.Guava from 32.0.1 to 31.1.0.8. Reference the package directly from the project to select a different version. 
 MyApp -> Plugin.Firebase 2.0.7 -> Plugin.Firebase.Firestore 2.0.5 -> Xamarin.Firebase.Firestore 124.8.1.1 -> Xamarin.Google.Guava (>= 32.0.1) 
 MyApp -> Xamarin.Google.Guava (>= 31.1.0.8)	MyApp	E:\My Projects\MAUI\MyApp-maui\MyApp\MyApp.csproj	1	

But the current version of `Xamarin.Google.Guava` is `31.1.0.8`, suggestion is also same.

**Comment by TobiasBuchholz** [comment]
So referencing the package directly in your project doesn't work? Or why can't you update it to version 32.0.1?

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I have updated the version of `Xamarin.Google.Guava` to `32.0.1` and installed `Plugin.Firebase 2.0.7`.

But still getting same list of errors when rebuilding the project in Release mode.

**Comment by sreejith-pagematics** [comment]
Any update on this?

**Comment by eyeveye** [comment]
> Well, the sample project of the plugin is building fine, also in release mode, so I don't know what your problem might be. If you provide a demo project I could try to look into it.

I encounter the same problem. To replicate it is easy, create a new Maui project, install Plugin.Firebase, press build in release mode OR publish, it will throw errors.
I found out it is due to the MAX PATH limit around 256 character limit, targeting IOS build. Enable long path solution in Windows does not help.

Unsure is due to Maui, or Firebase plugin or Visual studio bug.

**Comment by TobiasBuchholz** [comment]
> @TobiasBuchholz Could you please check this [demo](https://drive.google.com/file/d/12d-HgkNwY1CdkfUnqyWB15-5NMkA1GBX/view?usp=sharing) project?

Your demo project is building fine on my MacBook Pro with M1 Chip and VS for Mac Version 17.6.5 (build 417). Are you using Windows 11 for development?

> I encounter the same problem. To replicate it is easy, create a new Maui project, install Plugin.Firebase, press build in release mode OR publish, it will throw errors. I found out it is due to the MAX PATH limit around 256 character limit, targeting IOS build. Enable long path solution in Windows does not help.
> 
> Unsure is due to Maui, or Firebase plugin or Visual studio bug.

Sounds like your problem could be releated to issue https://github.com/TobiasBuchholz/Plugin.Firebase/issues/199.

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I am using windows 10 and my VS version is 17.7.6.

Did you try in release mode build? Are you able to achieve the APK file?

**Comment by TobiasBuchholz** [comment]
Yup, I made a release build, deployed the apk on a device and opened the app without any problems 🤷‍♂️

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz  Could you please check with this [demo](https://drive.google.com/file/d/1yCocAUawDlNZZ9ZjnT9iwmgBMIj4TXXI/view?usp=sharing) project? I have added all other NuGet packages into the same demo project that I shared with you previously.

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz 

I have installed Visual Studio 17.9.0 Preview version and in that also getting same list of errors. 
I removed the packages and commented all the firebase related codes, at that time I have no issue for creating the APK file and rebuilding the app in release mode. 
Can you do something on this issue? I am stuck on this for one week.

**Comment by TobiasBuchholz** [comment]
> @TobiasBuchholz Could you please check with this [demo](https://drive.google.com/file/d/1yCocAUawDlNZZ9ZjnT9iwmgBMIj4TXXI/view?usp=sharing) project? I have added all other NuGet packages into the same demo project that I shared with you previously.

On my MacBook I've donwloaded your project, went straight to building in release mode and deployed the apk to a real device, without any issues...if you do the same on your machine, are you getting the same error messages like on your original project?

Which features do you want to use from the library? Maybe your problem resolves itself when you just install the separate packages you need instead of the bundled `Plugin.Firebase` one.

**Comment by TobiasBuchholz** [comment]
Alright, then you can try just adding `Plugin.Firebase.CloudMessaging` instead of `Plugin.Firebase`, maybe this helps.

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I am using all the below using statements, if I only install `Plugin.Firebase.CloudMessaging` all those will show error.

![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/8adf87e6-f413-4248-947a-821daaf8d347)

**Comment by TobiasBuchholz** [comment]
Why are there references to `Auth`, `Crashlytics` and `DynamicLinks` when you only want to implement push notifications?

I've now setup my old windows pc (win 10) with the newest Visual Studio and checked out again your demo project in debug and release mode and got the same result as before, it's building fine.

What output do you get when running `dotnet --version` and `dotnet workload list` in the command line?

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz 
> Why are there references to Auth, Crashlytics and DynamicLinks when you only want to implement push notifications?

I am implementing push notification on MAUI application and when I added the required code it shows to add the `Auth, Crashlytics and DynamicLinks` references to solve the code errors.

> What output do you get when running dotnet --version and dotnet workload list in the command line?

![Capture](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/b59f5c4e-566a-446c-9615-29033f779a88)

**Comment by TobiasBuchholz** [comment]
> I am implementing push notification on MAUI application and when I added the required code it shows to add the Auth, Crashlytics and DynamicLinks references to solve the code errors.

If you only want to implement push notifications, there is no code required that references other packages like Auth, Crashlytics or DynamicLinks.

> What output do you get when running dotnet --version and dotnet workload list in the command line?

In your demo project you are targetting `.net7` but on your machine you have installed `.net8` with its workloads. So you should either target `.net8` in your projects like this
```c#
<TargetFrameworks>net8.0-android;net8.0-ios</TargetFrameworks>
```

or you can install `.net7` with its workloads on your machine (https://dotnet.microsoft.com/en-us/download/dotnet) so that running `dotnet --version` outputs something like `7.0.309` and running `dotnet workload list` ouputs something like
```
Installed Workload Id      Manifest Version       Installation Source
---------------------------------------------------------------------
wasm-tools                 7.0.12/7.0.100         SDK 7.0.300        
maui-android               7.0.96/7.0.100         SDK 7.0.300        
ios                        16.4.7107/7.0.100      SDK 7.0.300        
android                    33.0.95/7.0.100        SDK 7.0.300        
wasm-tools-net6            7.0.12/7.0.100         SDK 7.0.300   
```

Maybe this solves your problem.

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I tried the first solution, updated the target to 8.0 like below, but same errors.
> <TargetFrameworks>net8.0-android;net8.0-ios</TargetFrameworks>

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I need to try with installing .net7. When installing .net 7 do I need to uninstall .net 8?

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz I try to update the `Plugin.firebse` package to 2.0.9 but I am getting below errors:

> Severity	Code	Description	Project	File	Line	Suppression State
Error		Package restore failed. Rolling back package changes for 'MyProjectName'.				

> Severity	Code	Description	Project	File	Line	Suppression State
Error	NU1605	Warning As Error: Detected package downgrade: Xamarin.AndroidX.Browser from 1.6.0.2 to 1.4.0.3. Reference the package directly from the project to select a different version. 
 MyProjectName -> Plugin.Firebase 2.0.9 -> Plugin.Firebase.Auth 2.0.5 -> Xamarin.Firebase.Auth 122.2.0 -> Xamarin.AndroidX.Browser (>= 1.6.0.2) 
 MyProjectName -> Xamarin.AndroidX.Browser (>= 1.4.0.3)	MyProjectName	E:\My Projects\MAUI\MyProjectName-app-maui\MyProjectName\MyProjectName.csproj	1	

But `Xamarin.AndroidX.Browser` current version number is 1.4.0.3 and I can't install  the latest versions since it is blocked by project.

![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/0d87175e-f9bf-416d-8ef3-ac0224ed8b02)

**Comment by sreejith-pagematics** [comment]
@eyeveye  Are you able to resolve this error?

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz  I don't know why you can't reproduce this issue since everyone else are facing it.

Could you please check with [this](https://drive.google.com/file/d/1JLTDzqIhSt2Jxh05wBqC4MdwPNlEOyVf/view?usp=sharing) new demo project? I have created it in `VS 17.9.0 Preview 1.1- .NET 8` and added `Plugin.Firebase-2.0.9` and `Plugin.Firebase.Crashlytics-2.0.1`. I am able to run it into my device in debug mode and then I build it in release mode. Then I am getting same set of errors. 

> Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3027	Could not copy "C:\Users\SREEJITH\.nuget\packages\xamarin.firebase.ios.cloudfirestore\8.10.0.3\lib\net6.0-ios15.4\Firebase.CloudFirestore.resources\grpc.xcframework\ios-arm64_x86_64-simulator\grpc.framework\PrivateHeaders\src\core\ext\filters\client_channel\backup_poller.h" to "bin\Release\net8.0-ios\iossimulator-x64\Firebase.CloudFirestore.resources\grpc.xcframework\ios-arm64_x86_64-simulator\grpc.framework\PrivateHeaders\src\core\ext\filters\client_channel\backup_poller.h". Exceeded retry count of 10. Failed. 	FirebaseSample	C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	4911	

> Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3030	Could not copy the file "C:\Users\SREEJITH\.nuget\packages\xamarin.firebase.ios.cloudfirestore\8.10.0.3\lib\net6.0-ios15.4\Firebase.CloudFirestore.resources\grpc.xcframework\ios-arm64\grpc.framework\PrivateHeaders\src\core\ext\filters\workarounds\workaround_cronet_compression_filter.h" because it was not found.	FirebaseSample	C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	4911	

> Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3021	Unable to copy file "C:\Users\SREEJITH\.nuget\packages\xamarin.firebase.ios.cloudfirestore\8.10.0.3\lib\net6.0-ios15.4\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\endpoint\load_report.upb.h" to "bin\Release\net8.0-ios\iossimulator-x64\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\endpoint\load_report.upb.h". Could not find a part of the path 'bin\Release\net8.0-ios\iossimulator-x64\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\endpoint\load_report.upb.h'.	FirebaseSample	C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	4911	

> Severity	Code	Description	Project	File	Line	Suppression State
Error	MSB3021	Unable to copy file "C:\Users\SREEJITH\.nuget\packages\xamarin.firebase.ios.cloudfirestore\8.10.0.3\lib\net6.0-ios15.4\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64_x86_64-simulator\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\lds.upb.h" to "bin\Release\net8.0-ios\iossimulator-x64\Firebase.CloudFirestore.resources\grpcpp.xcframework\ios-arm64_x86_64-simulator\grpcpp.framework\PrivateHeaders\src\core\ext\upb-generated\envoy\api\v2\lds.upb.h". The specified path, file name, or both are too long. The fully qualified file name must be less than 260 characters, and the directory name must be less than 248 characters.	FirebaseSample	C:\Program Files\Microsoft Visual Studio\2022\Preview\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	4911	

![image_2023_12_01T13_23_26_322Z](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/008a2334-ad9d-4db5-bb9b-5dcfa8afc38d)
![image_2023_12_01T13_26_18_727Z](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/3508f27f-a994-4506-bc69-a2d7b03be98f)
![image_2023_12_01T13_24_18_115Z](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/118500242/f82de00e-67a4-4dbf-80a6-75f01367087a)

**Comment by sreejith-pagematics** [comment]
@TobiasBuchholz 

If I target only android then no errors and able to build the APK file. So is these issues are really due to the iOS platform? It is support for iOS, right?

**Comment by AlleSchonWeg** [comment]
I think the iOS problems have nothing to do with this project. This project depends on the xamarin ios firebase bindings: https://github.com/xamarin/GoogleApisForiOSComponents and other libs have the similar problems.
The bindings project looks dead. MS only maintain the android firebase bindings. I contacted MS people via twitter and github and asked them to update the bindings but i get no response. 
To fix these issues somebody must create a new ios firebase bindings project!

**Comment by AndyG74** [comment]
I have recreated some of the problems that have already been discussed here and solved them as follows.


Installing Plugin.Firebase via nuget manager fails.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/57a1ad50-5e34-4d64-beec-2dc7de20c6ac)


So I installed the package via the command line, the long path (>256 characters) doesn't seem to be a problem here.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/3886d36e-5401-47b2-9020-35680863b2a3)


During the installation, the xamarin.build.download (v0.11.0) package is also installed, although a more recent version is already available.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/4c3f4349-0195-4453-908a-c6cc370a3553)


So I explicitly installed this more recent version for my project, which at least made this error disappear.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/e36c1266-bc29-4e66-a173-3689e4f6a939)


Instead there was another error, which was probably due to a path that was too long (>256 characters).
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/912d210e-b9eb-4858-b14d-3e50383a8266)


Now I have deactivated the iOS app in order to only compile the Android app.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/a471a86a-9c8f-4476-92a3-e4200656a49e)


now another error was issued.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/08099113-c5cb-4c53-b86c-e073129c452d)


I was able to eliminate this by adding the following packages.
```
<ItemGroup Condition="'$(TargetFramework)' == 'net8.0-android'">
    <PackageReference Include="Xamarin.AndroidX.Core" Version="1.12.0.2" />
    <PackageReference Include="Xamarin.AndroidX.Collection" Version="1.3.0.1" />
    <PackageReference Include="Xamarin.AndroidX.Collection.Ktx" Version="1.3.0.1" />
    <PackageReference Include="Xamarin.AndroidX.Activity.Ktx" Version="1.8.0.1" />
</ItemGroup>
```


Now the Android app can be compiled.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/bf68de98-6abf-4274-9047-53e9cfeb5907)


now the question: how do you get a .net8 Maui iOS app running with firebase push messaging?

@TobiasBuchholz: Do you have the option to make the path shorter with your Plugin.Firebase?

@All: Can anyone confirm that there is no problem with the long paths in Win10, or that there is a workaround there?

ps: my system is Win11, VS 2022 17.8.3

**Comment by AndyG74** [comment]
Oh, nice, i have found the solution. follow the this guid for enable long pathnames for win10/11
https://www.thewindowsclub.com/how-to-enable-or-disable-win32-long-paths-in-windows-11-10

and then you can compile succsessful also your ios-app.
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/61390650/418da1dd-5170-48bc-b30d-b0e31e778792)

**Comment by TobiasBuchholz** [comment]
Cool, thank you very much @AndyG74 for providing all this information! :) Are there other people having these problems who can confirm the tips from @AndyG74 solve the problems?

**Comment by PPT-DGREEN-GH** [comment]
I just want to confirm I have long file paths enabled on my Win11 machine and I'm getting what I believe to be a file length error:
![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/141932136/b16c619e-763c-475d-bf6b-de3f177e4ebc)

**Comment by Quaybe** [comment]
I have nuget packages set to C:/n, the shortest folder name possible, and have long file paths enabled in Win10. I can no longer get xamarin.firebase.ios.cloudfirestore short enough to build. I cannot do a rebuild on my application at all.

![image](https://github.com/TobiasBuchholz/Plugin.Firebase/assets/99894209/ffd01435-c145-4d4e-abc3-c90beb00eaea)

**Comment by mizuhs** [comment]
I can confirm that having long file names enabled and having the nuget folder changed to C:\n doesn't necessarily make it work. 

Disabling iOS in the project file, as suggested by @AndyG74, actually solves the build problem getting rid of the 500+ errors in the output, but since I need iOS as a target platform, it's not really an effective way in this case.
As noted before, it all comes down to the iOS part of this very promising plugin. Too bad it seems to be something beyond what the plugin author can do since it's more OS related.

**Comment by tranb3r** [comment]
> I can confirm that having long file names enabled and having the nuget folder changed to C:\n doesn't necessarily make it work.
> Btw, I'm using Windows 10, not 11.

It should work. What is your error?
Did you make sure you've closed visual studio first, then removed all packages from nuget cache, then restored them from command line? Immediately after that, you should check if the files previously in error do exist or not, via explorer.

**Comment by plettb** [comment]
"It should work." 🤣 That's right up there with "it works on my machine." 🤣 Being a developer, I'm intimately familiar with this kind of scenario from the "supporting" side, but now I find myself on the "client" side of the equation. 😊

Since this thread seems to have gone cold without a resolution, I'll try to provide information to keep it going.  I'm running on Windows 11, with VS 2022 17.10.2 (i.e. both OS and VS completely up-to-date as of this writing).  I've resolved the "long name" issue with the registry setting, and have set "NUGET_PACKAGES" to "C:\N".

Here are the steps I take to recreate the error:
1. I create a completely clean, new Maui application, using .NET 8.
2. I remove all the other operating systems from the csproj, leaving only Android
3. I add "Plugin.Firebase" "Latest stable 3.0.0"
4. I try to "Build the solution"
5. I get the following build result:
`
1>------ Build started: Project: MauiApp1, Configuration: Debug Any CPU ------
1>MauiApp1 -> D:\Research\MauiApp1\MauiApp1\bin\Debug\net8.0-android\MauiApp1.dll
1>MSBUILD : java.exe error JAVA0000: Error in C:\N\xamarin.androidx.collection.jvm\1.3.0.1\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-jvm.jar:androidx/collection/ArrayMapKt.class:
1>MSBUILD : java.exe error JAVA0000: Type androidx.collection.ArrayMapKt is defined multiple times: C:\N\xamarin.androidx.collection.jvm\1.3.0.1\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-jvm.jar:androidx/collection/ArrayMapKt.class, C:\N\xamarin.androidx.collection.ktx\1.2.0.9\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-ktx.jar:androidx/collection/ArrayMapKt.class
1>MSBUILD : java.exe error JAVA0000: Compilation failed
1>MSBUILD : java.exe error JAVA0000: java.lang.RuntimeException: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: C:\N\xamarin.androidx.collection.jvm\1.3.0.1\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-jvm.jar
1>MSBUILD : java.exe error JAVA0000: androidx/collection/ArrayMapKt.class
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:135)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.main(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:5)
1>MSBUILD : java.exe error JAVA0000: Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete, origin: C:\N\xamarin.androidx.collection.jvm\1.3.0.1\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-jvm.jar:androidx/collection/ArrayMapKt.class
1>MSBUILD : java.exe error JAVA0000: 	at Version.fakeStackEntry(Version_8.2.33.java:0)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.T.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:5)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:82)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:32)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:31)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.b(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:2)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:42)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.b(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:13)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:40)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:122)
1>MSBUILD : java.exe error JAVA0000: 	... 1 more
1>MSBUILD : java.exe error JAVA0000: Caused by: com.android.tools.r8.utils.b: Type androidx.collection.ArrayMapKt is defined multiple times: C:\N\xamarin.androidx.collection.jvm\1.3.0.1\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-jvm.jar:androidx/collection/ArrayMapKt.class, C:\N\xamarin.androidx.collection.ktx\1.2.0.9\buildTransitive\net6.0-android31.0\..\..\jar\androidx.collection.collection-ktx.jar:androidx/collection/ArrayMapKt.class
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.Q2.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:21)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.D2.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:54)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.D2.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:10)
1>MSBUILD : java.exe error JAVA0000: 	at java.base/java.util.concurrent.ConcurrentHashMap.merge(ConcurrentHashMap.java:2056)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.D2.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:6)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.graph.m4$a.d(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:6)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.dex.c.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:61)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.dex.c.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:12)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.dex.c.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:9)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:45)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.d(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:17)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.D8.c(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:69)
1>MSBUILD : java.exe error JAVA0000: 	at com.android.tools.r8.utils.S0.a(R8_8.2.33_429c93fd24a535127db6f4e2628eb18f2f978e02f99f55740728d6b22bef16dd:28)
1>MSBUILD : java.exe error JAVA0000: 	... 6 more
1>MSBUILD : java.exe error JAVA0000:
1>Done building project "MauiApp1.csproj" -- FAILED.
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
========== Build completed at 4:03 PM and took 16.958 seconds ==========
`

That's about as clean and simple as it can get.  In fact given how clean and simple this is, I can't help but wonder if anyone has actually successfully used "Plugin.Firebase" "Latest stable 3.0.0" from scratch.

Any help would be greatly appreciated.

**Comment by AdamEssenmacher** [comment]
@plettb try following the setup instructions in the project Readme. If that doesn't help, please link a sample repro project.

**Comment by AdamEssenmacher** [comment]
@plettb were you able to get it working?

**Comment by plettb** [comment]
@AdamEssenmacher this may have been a case of RTFM.  I think I'd missed adding the "Android specifics" (as per https://github.com/TobiasBuchholz/Plugin.Firebase?tab=readme-ov-file#android-specifics).

Unfortunately I can't recreate the problem this morning (running into other problems) and don't have the time atm.  I will try to revisit when I get a chance, but for now I think you can consider the problem I raised above to be resolved.

**Comment by AdamEssenmacher** [comment]
Closing the issue since it seems the questions here have been answered, and the discussion has really creeped in scope. If anyone is still having issues:

1. Read the README :)
2. Create a new issue, ideally with a repro project linked
